build: mimalloc.dll.lib is trying to be installed for MinGW

while building 2.2.2 release, I've encountered this issue
```
  -- Install configuration: "Release"
  -- Installing: D:/_/B/pkg/mingw-w64-ucrt-x86_64-mimalloc/ucrt64/lib/libmimalloc.dll.dll.a
  -- Installing: D:/_/B/pkg/mingw-w64-ucrt-x86_64-mimalloc/ucrt64/bin/libmimalloc.dll
  -- Installing: D:/_/B/pkg/mingw-w64-ucrt-x86_64-mimalloc/ucrt64/lib/cmake/mimalloc/mimalloc.cmake
  -- Installing: D:/_/B/pkg/mingw-w64-ucrt-x86_64-mimalloc/ucrt64/lib/cmake/mimalloc/mimalloc-release.cmake
  CMake Error at build-UCRT64/cmake_install.cmake:78 (file):
    file INSTALL cannot find "D:/_/B/src/build-UCRT64/mimalloc.dll.lib": File
    exists.
```
that's because install target is hardcoded for mimalloc.dll.lib for all Windows targets: https://github.com/microsoft/mimalloc/blob/660d749d77822e54b77acecf82f1aa8f348625ae/CMakeLists.txt#L587

as seen above, import library is already installed (but still with wrong extension)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -584,7 +584,7 @@ if(MI_BUILD_SHARED)
   install(TARGETS mimalloc EXPORT mimalloc ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
   install(EXPORT mimalloc DESTINATION ${mi_install_cmakedir})
 
-  if(WIN32)
+  if(WIN32 AND NOT MINGW)
     # On windows, the import library name for the dll would clash with the static mimalloc.lib library
     # so we postfix the dll import library with `.dll.lib` (and also the .pdb debug file)
     set_property(TARGET mimalloc PROPERTY ARCHIVE_OUTPUT_NAME "${mi_libname}.dll" )

